src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -772,7 +773,7 @@ void SpatialPooler::updatePermanencesForColumn_(vector<Real>& perm,
   }
 
   numConnected = 0;
-  for (UInt i = 0; i < perm.size(); ++i)
+  for (UInt i = 0; i < perm.size(); i++)
```

Comment:
@scottpurdy can you confirm this is correct? If so, it might be one of the causes of https://github.com/numenta/nupic/issues/1352